frontend/js/src/utils/utils.tsx

```diff
@@ -131,6 +131,61 @@ const searchForYoutubeTrack = async (
   return null;
 };
 
+const searchForSoundcloudTrack = async (
+  soundcloudToken?: string,
+  trackName?: string,
+  artistName?: string,
+  releaseName?: string
+): Promise<string | null> => {
+  if (!soundcloudToken) return null;
+  let query = trackName ?? "";
+  if (artistName) {
+    query += ` ${artistName}`;
+  }
+  // Considering we cannot tell the Soundcloud API that this should match only an album title,
+  // results are paradoxically sometimes worse if we add it to the query
```

Comment:
These comments were for the youtube search. I haven't evaluated the SoundCloud search, but maybe this doesn't hold true for it.